### PR TITLE
feat(migration): #MED-327 Vertx migration

### PR DESCRIPTION
**Pull request by Glama107**
## Describe your changes

Migration de Vertx 3 vers Vertx 4

## Issue ticket number and link

https://jira.support-ent.fr/browse/MED-327

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java [hunk 1]
```diff
@@ -160,15 +132,66 @@ public void get(String userId, String structure, Handler<Either<String, JsonArra
...
+                     beautifyRessourcesResult(promise, uai, structureName, resources);
+                 });
+                 response.exceptionHandler(throwable -> {
+                         log.error("[DefaultResourceService@get] failed to get GAR response: " + throwable.getMessage());
+                         promise.fail(throwable.getMessage());
+                 });
+             }
+         })
+         .onFailure(err -> {
+             log.error("" + err.getCause().getMessage());
```
Comment (Mattox0) [inline src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java]: Format des erreurs : `log.error("[DefaultResourceService@get] [ton erreur];` pour rester iso avec toutes les erreurs

**Review by Mattox0 — APPROVED** [review]

File: src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java (hunk 1 above)
Comment (Glama107) [reply src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java]: C'est fait !

**Review by Mattox0 — APPROVED** [review]

File: src/main/java/fr/openent/gar/export/impl/ExportImpl.java [hunk 2]
```diff
@@ -120,7 +120,7 @@ private void exportAndSend(String entId, final String source, Handler<String> ha
                         String n = (String) vertx.sharedData().getLocalMap("server").get("node");
                         String node = (n != null) ? n : "";
 
-                        eb.send(node + "sftp", sendTOGar, new DeliveryOptions().setSendTimeout(300 * 1000L),
+                        eb.request(node + "sftp", sendTOGar, new DeliveryOptions().setSendTimeout(300 * 1000L),
                                 handlerToAsyncHandler((Message<JsonObject> messageResponse) -> {
                             if (messageResponse.body().containsKey("status") && messageResponse.body().getString("status").equals("error")) {
                                 String e = "Send to GAR tar GZ by sftp but received an error : " +
```
Comment (qperie) [inline src/main/java/fr/openent/gar/export/impl/ExportImpl.java:126]: Quali des logs

File: src/main/java/fr/openent/gar/export/impl/ExportImpl.java [hunk 3]
```diff
@@ -133,7 +133,7 @@ private void exportAndSend(String entId, final String source, Handler<String> ha
                                 sendTOGar
                                         .put("local-file", FileUtils.appendPath(exportArchivePath, md5File))
                                         .put("dist-file", FileUtils.appendPath(tenant.getString("dir-dest"), md5File));
-                                eb.send(node + "sftp", sendTOGar, handlerToAsyncHandler(message1 -> {
+                                eb.request(node + "sftp", sendTOGar, handlerToAsyncHandler(message1 -> {
                                     if (message1.body().containsKey("status") && message1.body().getString("status").equals("error")) {
                                         String e = "FAILED Send to Md5 by sftp : " +
                                                 messageResponse.body().getString("message");
```
Comment (qperie) [inline src/main/java/fr/openent/gar/export/impl/ExportImpl.java:139]: Quali des logs

File: src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java [hunk 4]
```diff
@@ -110,45 +113,14 @@ public void get(String userId, String structure, Handler<Either<String, JsonArra
...
-                            .putHeader("Cache-Control", "no-cache")
-                            .putHeader("Date", new Date().toString());
-
-                    clientRequest.end();
-                } else{
+
+                    garRequest(httpClient, resourcesUri, garHostNoProtocol, uai, structureName)
+                        .onSuccess(result -> handler.handle(new Either.Right<>(result)))
+                        .onFailure(err -> {
+                            log.error("[DefaultResourceService@get] An error occurred when fetching structure UAI");
```
Comment (qperie) [inline src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java:120]: Quali des logs

File: src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java [hunk 5]
```diff
@@ -160,15 +132,66 @@ public void get(String userId, String structure, Handler<Either<String, JsonArra
...
+                        .add("Cache-Control", "no-cache")
+                        .add("Date", new Date().toString())
+                );
+
+         httpClient.request(requestOptions)
+         .flatMap(HttpClientRequest::send)
+         .onSuccess(response -> {
+             if (response.statusCode() != 200) {
+                 log.error("try to call " + resourcesUri);
+                 log.error(response.statusCode() + " " + response.statusMessage());
```
Comment (qperie) [inline src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java:152]: Quali des logs

File: src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java [hunk 6]
```diff
@@ -160,15 +132,66 @@ public void get(String userId, String structure, Handler<Either<String, JsonArra
...
+                 });
+             } else {
+                 Buffer responseBuffer = new BufferImpl();
+                 response.handler(responseBuffer::appendBuffer);
+                 response.endHandler(aVoid -> {
+                     JsonObject resources = new JsonObject(decompress(responseBuffer));
+                     beautifyRessourcesResult(promise, uai, structureName, resources);
+                 });
+                 response.exceptionHandler(throwable -> {
+                         log.error("[DefaultResourceService@get] failed to get GAR response: " + throwable.getMessage());
```
Comment (qperie) [inline src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java:174]: Quali des logs

File: src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java [hunk 7]
```diff
@@ -160,15 +132,66 @@ public void get(String userId, String structure, Handler<Either<String, JsonArra
...
+                         promise.fail(throwable.getMessage());
+                 });
+             }
+         })
+         .onFailure(err -> {
+             log.error("[DefaultResourceService@get] failed to request :" + err.getCause().getMessage());
```
Comment (qperie) [inline src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java:180]: Quali des logs

File: src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java [hunk 8]
```diff
@@ -190,7 +213,7 @@ private String decompress(Buffer buffer) {
     private HttpClient generateHttpClient(final URI uri, final String certPath, final String keyPath) {
         HttpClientOptions options = new HttpClientOptions()
                 .setDefaultHost(uri.getHost())
-                .setDefaultPort(uri.getPort() == -1 ? ("https".equals(uri.getScheme()) ? 443 : 80) : uri.getPort())
+                .setDefaultPort("https".equals(uri.getScheme()) ? 443 : 80)
```
Comment (qperie) [inline src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java:216]: Régression, ce code n'aurait pas du être modifié

**Pull request merged**